### ISSUE TITLE
Document how state works with `merge` and `nest`

### ISSUE THREAD

**Issue by davidpdrsn** [tokio-rs/axum#1313]
Gotten a few questions about how states work with `merge` and `nest`. Things like whether the state needs to be passed to each router and whether the state of the inner routers is dropped. This is something we should document.

**Comment by genusistimelord** [comment]
Yeah I noticed this when using AxumFlash and passing the config as the State. I had to change my Routing functions to take in the State to set them individually. 

```
Router::with_state(flash_config.clone())
        .route("/", get(root))
       .merge(main_routes(flash_config.clone()))
```

which results in 
```
pub fn main_routes(config: axum_flash::Config) -> Router<axum_flash::Config> {
    Router::with_state(config)
        .route("/main", get(main_page))
}
```

**Comment by davidpdrsn** [comment]
https://github.com/tokio-rs/axum/pull/1368 is making things more flexible. When that is in I'll write some docs for it.

**Comment by davidpdrsn** [comment]
I think we're in a good place in terms for docs. We can expand things when people ask questions but for now I'll close this.

**Comment by davidpdrsn** [comment]
Good idea. Do you wanna make a pr?
